### PR TITLE
Fixing panic when pubsub handler returns an error

### LINKED ISSUE #3281: Dapr runtime panic when handle Pub/Sub without CloudEvents  status retry or drop
## In what area(s)?

/area runtime

## What version of Dapr?

1.2.0

panic log

```
panic: interface conversion: interface {} is nil, not string

goroutine 106 [running]:
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).publishMessageHTTP(0xc0001e0d80, 0x4998520, 0xc0000b9800, 0xc000a0a740, 0x44df401, 0xc000a0a740)
	/Users/runner/work/dapr/dapr/pkg/runtime/runtime.go:1222 +0x163f
github.com/dapr/dapr/pkg/runtime.(*DaprRuntime).beginPubSub.func1(0x4998520, 0xc0000b9800, 0xc00092edd0, 0x4998520, 0xc0000b9800)
	/Users/runner/work/dapr/dapr/pkg/runtime/runtime.go:487 +0x39d
github.com/dapr/components-contrib/pubsub/redis.(*redisStreams).processMessage(0xc0009fac60, 0xc00078a960, 0xf, 0xc00078a990, 0x13, 0x18, 0xc0006cb0e0, 0xf, 0x0, 0xc000eac240, ...)
	/Users/runner/go/pkg/mod/github.com/dapr/components-contrib@v1.2.0/pubsub/redis/redis.go:475 +0x1bf
github.com/dapr/components-contrib/pubsub/redis.(*redisStreams).worker(0xc0009fac60)
	/Users/runner/go/pkg/mod/github.com/dapr/components-contrib@v1.2.0/pubsub/redis/redis.go:458 +0xd9
created by github.com/dapr/components-contrib/pubsub/redis.(*redisStreams).Init
	/Users/runner/go/pkg/mod/github.com/dapr/components-contrib@v1.2.0/pubsub/redis/redis.go:373 +0x3f3
```

related source code:

https://github.com/dapr/dapr/blob/8e960e2632d435ccc4cf1cd1d2f8446784e0c39a/pkg/runtime/runtime.go#L1220

rawPayload not always have the `pubsub.IDField(id)` field

### PR DESCRIPTION
**Pull request by pkedy**
# Description

This PR removes some explicit casts to `string` when logging after a pubsub handler returns an error. Since all the formats accept the cloud event values as `%v` there is no need to assume they are strings.

## Issue reference

Resolves #3281

## Checklist

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by wcs1only** [comment]
Can we add a test that reproduces the bug?

**Comment by pkedy** [comment]
@wcs1only I added a test but also noticed this only fixes the HTTP side.  The gRPC pubsub needs the same sort of fix. I'll add to this PR next week.

**Comment by artursouza** [comment]
LGTM

File: pkg/channel/testing/grpc_mock_client_conn.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package testing
```
Comment (artursouza) [inline pkg/channel/testing/grpc_mock_client_conn.go:1]: Pardon my ignorance here but does Go compiler ignore this automatically for the main binary?

Comment (pkedy) [reply pkg/channel/testing/grpc_mock_client_conn.go:1]: Yes, only imported packages are included in the binary.

**Review by artursouza — APPROVED** [review]

**Pull request merged**